src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.LazySyntaxTree.cs

```diff
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    public partial class CSharpSyntaxTree
+    {
+        private class LazySyntaxTree : CSharpSyntaxTree
+        {
+            private readonly CSharpParseOptions _options;
+            private readonly string _path;
+            private readonly SourceText _text;
+            private readonly ImmutableDictionary<string, ReportDiagnostic> _diagnosticOptions;
+            private CSharpSyntaxNode? _lazyRoot;
+
+            internal LazySyntaxTree(
+                SourceText text,
+                string path,
+                CSharpParseOptions options,
+                ImmutableDictionary<string, ReportDiagnostic>? diagnosticOptions)
+            {
+                Debug.Assert(options != null);
```

Comment:
Curious: why assert this but not `text`?

Reply:
I have no idea lol. Looks like a few consistency fixes could be made here.